crates/rolldown_binding/src/options/binding_output_options/binding_chunk_file_names.rs

```diff
@@ -0,0 +1,65 @@
+use anyhow::anyhow;
+use napi::{bindgen_prelude::FromNapiValue, type_of, Either, ValueType};
+use rolldown::ChunkFileNamesOutputOption;
+use serde::Deserialize;
+
+use crate::types::{binding_pre_rendered_chunk::PreRenderedChunk, js_callback::JsCallback};
+
+pub struct BindingChunkFileNamesOutputOption(ChunkFileNamesOutputOption);
+
+impl From<ChunkFileNamesOutputOption> for BindingChunkFileNamesOutputOption {
+  fn from(value: ChunkFileNamesOutputOption) -> Self {
+    Self(value)
+  }
+}
+
+impl From<BindingChunkFileNamesOutputOption> for ChunkFileNamesOutputOption {
+  fn from(val: BindingChunkFileNamesOutputOption) -> Self {
+    val.0
+  }
+}
+
+impl FromNapiValue for BindingChunkFileNamesOutputOption {
```

Comment:
We have a discussion about these part. Could you try use `Either<String, JsFunction>` instead of this?

We want to avoid manually writing `FromNapiValue` as much as possible.

Reply:
I tried it before, but it didn't work. 
related issue https://github.com/napi-rs/napi-rs/issues/2198

Reply:
See how threadsafe function created in https://napi.rs/docs/concepts/threadsafe-function.

Notice what I type is `Either<String, JsFunction>`.

Reply:
Sorry. And I also tried JsFunction, it was deprecated
https://github.com/napi-rs/napi-rs/blob/327998d0aca2302e516fb2f06ee9ca59f8fc2799/crates/napi/src/js_values/function.rs#L13

Reply:
Didn't thought of that. Well you could just follow the suggestion there:

https://github.com/napi-rs/napi-rs/blob/327998d0aca2302e516fb2f06ee9ca59f8fc2799/examples/napi/src/function.rs#L113-L128

Reply:
> Didn't thought of that. Well you could just follow the suggestion there:
> 
> https://github.com/napi-rs/napi-rs/blob/327998d0aca2302e516fb2f06ee9ca59f8fc2799/examples/napi/src/function.rs#L113-L128

Ok, I will try it in next commit

Reply:
Use `FunctionRef`, which doesn't require lifetime.

Reply:
Done